dev/managedservicesplatform/spec/monitoring.go

```diff
@@ -89,11 +111,93 @@ func (r *ResponseCodeRatioAlertSpec) Validate() []error {
 	}
 
 	if r.Duration != nil {
-		duration, err := time.ParseDuration(*r.Duration)
-		if err != nil {
-			errs = append(errs, errors.Wrap(err, "responseCodeRatios[].duration must be in the format of XXs"))
-		} else if duration%time.Minute != 0 {
-			errs = append(errs, errors.New("responseCodeRatios[].duration must be a multiple of 60s"))
+		if *r.Duration < 0 {
+			errs = append(errs, errors.New("responseCodeRatios[].duration must be a whole number"))
+		}
+		if *r.Duration > 1440 { // 24 hours
+			errs = append(errs, errors.New("responseCodeRatios[].duration must be less than 1440 minutes"))
+		}
+	}
+
+	return errs
+}
+
+type CustomAlertQueryType string
+
+const (
+	MQL    CustomAlertQueryType = "MQL"
+	PromQL CustomAlertQueryType = "PromQL"
```

Comment:
It's verbose, but Go conventions would dictate this naming scheme:

```suggestion
	CustomAlertQueryTypeMQL    CustomAlertQueryType = "MQL"
	CustomAlertQueryTypePromQL CustomAlertQueryType = "PromQL"
```

Reply:
I also suggested lowercase for the values because in our spec, all the other "type" configs are all-lowercase